File: solutions/swift/queen-attack/1/Sources/QueenAttack/QueenAttack.swift
enum QueenError: Error { case inValidRow, inValidColumn }
class Queen {
  let column, row: Int
  init(row: Int, column: Int) throws {
    guard (0...7).contains(row) else { throw QueenError.inValidRow }
    guard (0...7).contains(column) else { throw QueenError.inValidColumn }
    (self.row, self.column) = (row, column)
  }
  func canAttack(other: Queen) -> Bool {
    return column == other.column || row == other.row
      || abs(row - other.row) == abs(column - other.column)
  }
}

File: solutions/lua/queen-attack/2/queen-attack.lua
return function(pos)
  assert(pos.row >= 0 and pos.row <= 7 and pos.column >= 0 and pos.column <= 7)
  pos.can_attack = function(other)
    return pos.column == other.column or pos.row == other.row or
        math.abs(pos.row - other.row) == math.abs(pos.column - other.column)
  end
  return pos
end
